src/ChannelServer/Network/ChannelClient.cs

```diff
@@ -104,9 +104,13 @@ public Creature GetSummonedPetSafe(long id)
 		/// </summary>
 		public override void CleanUp()
 		{
-			// Dispose creatures, to remove subscriptions and stuff.
-			// Do this before unspawning, the creature might need the region.
-			foreach (var creature in this.Creatures.Values)
+
+            // Moved here to always be called when a client is being killed off, on the suggestion of E.
+            ChannelServer.Instance.Events.OnPlayerDisconnect(Controlling);
```

Comment:
I was actually thinking of adding new events and putting this one back in disconnection request, because it's quite likely someone will want to know about players actually logging out, or channel changing, instead of the catch all "this client is being killed". (Which will be an event of some kind too)